client-tez/src/main/java/org/apache/tez/dag/app/RssDAGAppMaster.java

```diff
@@ -0,0 +1,363 @@
+import org.apache.tez.common.TezClassLoader;
+import org.apache.tez.common.TezCommonUtils;
+import org.apache.tez.common.TezUtilsInternal;
+import org.apache.tez.common.VersionInfo;
+import org.apache.tez.dag.api.TezConfiguration;
+import org.apache.tez.dag.api.TezConstants;
+import org.apache.tez.dag.api.TezException;
+import org.apache.tez.dag.api.records.DAGProtos;
+import org.apache.tez.dag.api.records.DAGProtos.AMPluginDescriptorProto;
+import org.apache.tez.dag.library.vertexmanager.ShuffleVertexManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.client.api.ShuffleWriteClient;
+
+import static org.apache.tez.common.TezCommonUtils.TEZ_SYSTEM_SUB_DIR;
+
+public class RssDAGAppMaster extends DAGAppMaster {
+  private static final Logger LOG = LoggerFactory.getLogger(RssDAGAppMaster.class);
+  private ShuffleWriteClient shuffleWriteClient;
+  private TezRemoteShuffleManager tezRemoteShuffleManager;
+  private final String rssConfFileLocalResourceName = "rss_conf.xml";
+  private DAGProtos.PlanLocalResource rssConfFileLocalResource;
+  final ScheduledExecutorService scheduledExecutorService = Executors.newSingleThreadScheduledExecutor(
+    new ThreadFactory() {
+      @Override
+      public Thread newThread(Runnable r) {
+        Thread t = Executors.defaultThreadFactory().newThread(r);
+        t.setDaemon(true);
+        return t;
+      }
+    }
+  );
+
+  public RssDAGAppMaster(ApplicationAttemptId applicationAttemptId,
+                         ContainerId containerId, String nmHost, int nmPort, int nmHttpPort,
+                         Clock clock, long appSubmitTime, boolean isSession, String workingDirectory,
+                         String [] localDirs, String[] logDirs, String clientVersion,
+                         Credentials credentials, String jobUserName, AMPluginDescriptorProto pluginDescriptorProto) {
+    super(applicationAttemptId, containerId, nmHost, nmPort, nmHttpPort, clock, appSubmitTime, isSession,
+            workingDirectory, localDirs, logDirs, clientVersion, credentials, jobUserName, pluginDescriptorProto);
+  }
+
+  public ShuffleWriteClient getShuffleWriteClient() {
+    return shuffleWriteClient;
+  }
+
+  public void setShuffleWriteClient(ShuffleWriteClient shuffleWriteClient) {
+    this.shuffleWriteClient = shuffleWriteClient;
+  }
+
+  public TezRemoteShuffleManager getTezRemoteShuffleManager() {
+    return tezRemoteShuffleManager;
+  }
+
+  public void setTezRemoteShuffleManager(TezRemoteShuffleManager tezRemoteShuffleManager) {
+    this.tezRemoteShuffleManager = tezRemoteShuffleManager;
+  }
+
+  /**
+   * Init and Start Rss Client
+   * @param appMaster
+   * @param conf
+   * @param applicationAttemptId
+   * @throws Exception
+   */
+  public static void initAndStartRSSClient(final RssDAGAppMaster appMaster, Configuration conf,
+                                           ApplicationAttemptId  applicationAttemptId) throws Exception {
```

Comment:
nit: pay attention to indentation, 4 characters.

Reply:
I think you should load the checkstyle file into IDEA .

Reply:
> 

checkstyle can not find this problem,  Style checking has been done before submitting the code